lms/djangoapps/courseware/grades.py

```diff
@@ -152,6 +223,17 @@ def _grade(student, request, course, keep_raw_scores):
 
     More information on the format is in the docstring for CourseGrader.
     """
+    if field_data_cache is None:
+        with manual_transaction():
+            field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
+                course.id, student, course, depth=None, descriptor_filter=descriptor_affects_grading
+            )
+
+    max_scores_cache = MaxScoresCache(course.subtree_edited_on.isoformat())
```

Comment:
Yup, that's the idea. I set the timeout for a day otherwise, just in case something gets borked. Still need to add tests around this behavior, but it seems to work.